File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    inorderTraversal(root) {
        const inOrder = [];
        const _dfs = (node) => {
            if (!node) return 
            _dfs(node.left);
            inOrder.push(node.val);
            _dfs(node.right);
        }
        _dfs(root);
        return inOrder;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = left = right = 0
        while right < len(prices):
            profit = prices[right] - prices[left]
            max_profit = max(max_profit, profit)
            if profit <= 0:
                left = right
            
            right += 1
        
        return max_profit